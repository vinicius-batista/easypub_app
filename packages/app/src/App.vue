<template>
  <v-app style="background-color: #f1f3f5">
    <v-container :style="{ padding: 0 }">
      <router-view></router-view>
    </v-container>
  </v-app>
</template>

<script>
export default {}
</script>
