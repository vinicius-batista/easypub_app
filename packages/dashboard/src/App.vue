<template>
  <v-app style="background-color: #f1f3f5">
    <router-view></router-view>
  </v-app>
</template>

<script>
export default {
  name: 'App',
}
</script>
