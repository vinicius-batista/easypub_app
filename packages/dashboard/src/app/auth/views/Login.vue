<template>
  <AuthForm
    buttonText="Login"
    helperText="NÃ£o possui uma conta?"
    actionText="Cadastre-se agora!"
    @action-text:clicked="changeRoute('auth.register')"
    data-testId="login-page"
  >
    <ValidationProvider
      v-for="{ label, model, icon, type, rules } in form"
      :key="label"
      v-bind="{ rules }"
      :name="label"
    >
      <template v-slot="{ errors }">
        <v-text-field
          class="my-2"
          :type="type"
          :label="label"
          v-model="input[model]"
          :prepend-icon="icon"
          :error-messages="errors"
          filled
          :data-testId="model"
        />
      </template>
    </ValidationProvider>
  </AuthForm>
</template>

<script>
import AuthForm from '../components/AuthForm'
import { ValidationProvider } from 'vee-validate'

export default {
  name: 'Login',
  components: { AuthForm, ValidationProvider },
  data: () => ({
    input: {
      email: '',
      password: '',
    },
    form: [
      {
        label: 'Email',
        model: 'email',
        icon: 'fa-envelope',
        rules: 'required|email',
      },
      {
        label: 'Senha',
        model: 'password',
        icon: 'fas fa-lock',
        type: 'password',
        rules: 'required|min:6',
      },
    ],
  }),
  methods: {
    changeRoute(name) {
      this.$router.push({ name })
    },
  },
}
</script>
