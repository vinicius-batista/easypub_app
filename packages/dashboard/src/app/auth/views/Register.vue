<template>
  <AuthForm
    buttonText="Cadastre-se"
    helperText="Já possui uma conta?"
    actionText="Entre agora!"
    data-testId="register-page"
    @action-text:clicked="changeRoute('auth.login')"
  >
    <ValidationProvider
      v-for="{ label, icon, type, rules, model } in form"
      :key="label"
      v-bind="{ rules }"
      :name="label"
    >
      <template v-slot="{ errors }">
        <v-text-field
          class="my-2"
          :type="type"
          :label="label"
          v-model="input[model]"
          :prepend-icon="icon"
          :error-messages="errors"
          filled
          :data-testId="model"
        />
      </template>
    </ValidationProvider>
    <ValidationProvider rules="required|size:2000000" name="Foto do bar">
      <template v-slot="{ errors }">
        <v-file-input
          class="my-2"
          label="Foto do bar"
          accept="image/png, image/jpeg, image/bmp"
          v-model="input['barAvatar']"
          prepend-icon="fa-image"
          :error-messages="errors"
          filled
          data-testId="barAvatar"
        />
      </template>
    </ValidationProvider>
  </AuthForm>
</template>

<script>
import AuthForm from '../components/AuthForm'
import { ValidationProvider } from 'vee-validate'

export default {
  name: 'Register',
  components: { AuthForm, ValidationProvider },
  data: () => ({
    input: {
      email: '',
      password: '',
      phone: '',
      name: '',
      barName: '',
      barAddress: '',
      barAvatar: null,
    },
    form: [
      {
        label: 'Nome',
        model: 'name',
        icon: 'fa-user',
        rules: 'required|alpha_spaces',
      },
      {
        label: 'Email',
        model: 'email',
        icon: 'fa-envelope',
        rules: 'required|email',
      },
      {
        label: 'Telefone',
        model: 'phone',
        icon: 'fa-phone',
        rules: 'required',
        mask: '(##) ##### - ####',
        type: 'tel',
      },
      {
        label: 'Senha',
        model: 'password',
        icon: 'fas fa-lock',
        type: 'password',
        rules: 'required|min:6',
      },
      {
        label: 'Nome do bar',
        model: 'barName',
        icon: 'fa-beer',
        rules: 'required',
      },
      {
        label: 'Endereço do bar',
        model: 'barAddress',
        icon: 'fas fa-map-marker-alt',
        rules: 'required',
      },
    ],
  }),
  methods: {
    changeRoute(name) {
      this.$router.push({ name })
    },
  },
}
</script>
