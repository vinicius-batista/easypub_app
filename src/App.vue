<template>
  <v-app style="background-color: #f3f7f9">
    <v-content>
      <v-container grid-list-xs :style="{ padding: 0 }">
        <router-view></router-view>
      </v-container>
    </v-content>
  </v-app>
</template>
