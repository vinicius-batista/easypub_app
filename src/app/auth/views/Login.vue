<template>
  <div>
    <LogoCard title="Login" />
    <FormCard
      helperText="NÃ£o possui uma conta?"
      actionText="Cadastre-se agora!"
      @actionClick="changeRoute('auth.register')"
    >
      <FormErrorMessage ref="formErrorMessage" />
      <ApolloMutation
          :mutation="$options.loginUserMutation"
          @done="submitSuccess"
          @error="handleError"
        >
          <template slot-scope="{ mutate, loading }">
            <v-form v-model="valid" @submit.prevent="mutate({ variables: { input } })">
              <v-text-field
                class="my-2"
                v-for="{ label, model, icon, type, rules } in form"
                v-validate="rules"
                :data-vv-name="model"
                :data-vv-as="label.toLowerCase()"
                :type="type"
                :key="model"
                :label="label"
                v-model="input[model]"
                :prepend-icon="icon"
                :error-messages="errors.collect(model)"
                box
              />
              <h6
                class="tertiary--text body-2 font-weight-medium mb-3"
                @click="changeRoute('auth.register')"
                style="cursor: pointer;"
              >
                Esqueceu sua senha?
              </h6>
              <SendButton
                text="Acessar"
                v-bind="{loading}"
                :disabled="!valid"
              />
            </v-form>
          </template>
      </ApolloMutation>
    </FormCard>
  </div>
</template>

<script>
import LogoCard from '../components/LogoCard'
import FormCard from '../components/FormCard'
import FormErrorMessage from '@/components/FormErrorMessage'
import SendButton from '@/components/SendButton'
import { getData } from '@/helpers/graphql'
import { mapActions } from 'vuex'
import { loginUserMutation } from '@/domains/auth/graphql'

export default {
  name: 'Login',
  components: {
    LogoCard,
    FormCard,
    FormErrorMessage,
    SendButton
  },
  loginUserMutation,
  data: () => ({
    valid: false,
    input: {
      email: '',
      password: ''
    },
    form: [
      {
        label: 'Email',
        model: 'email',
        icon: 'fa-envelope',
        rules: 'required|email'
      },
      {
        label: 'Senha',
        model: 'password',
        icon: 'fa-lock',
        type: 'password',
        rules: 'required|min:6'
      }
    ]
  }),
  methods: {
    ...mapActions('auth', ['setTokens']),
    changeRoute (name) {
      this.$router.push({ name })
    },
    submitSuccess (result) {
      return Promise
        .resolve(result)
        .then(getData('loginUser'))
        .then(this.setTokens)
        .then(() => this.changeRoute('home.bars'))
    },
    handleError (error) {
      this.$refs.formErrorMessage.handleError(error)
    }
  }
}
</script>
