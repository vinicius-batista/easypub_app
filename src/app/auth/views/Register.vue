<template>
  <div>
    <LogoCard title="Cadastro" />
    <FormCard
      helperText="JÃ¡ possui uma conta?"
      actionText="Entre agora!"
      @actionClick="changeRoute('auth.login')"
    >
      <FormErrorMessage ref="formErrorMessage" />
      <ApolloMutation
          :mutation="$options.registerUserMutation"
          @done="submitSuccess"
          @error="handleError"
        >
          <template slot-scope="{ mutate, loading }">
            <v-form v-model="valid" @submit.prevent="mutate({ variables: { input } })">
              <v-text-field
                class="my-2"
                v-for="{ label, model, icon, type, rules, mask } in form"
                v-validate="rules"
                :data-vv-name="model"
                :data-vv-as="label.toLowerCase()"
                :key="model"
                :label="label"
                v-model="input[model]"
                :prepend-icon="icon"
                :error-messages="errors.collect(model)"
                v-bind="{ type, mask }"
                box
              />
              <SendButton
                text="Enviar"
                v-bind="{loading}"
                :disabled="!valid"
              />
            </v-form>
          </template>
      </ApolloMutation>
    </FormCard>
  </div>
</template>

<script>
import LogoCard from '../components/LogoCard'
import FormCard from '../components/FormCard'
import FormErrorMessage from '@/components/FormErrorMessage'
import SendButton from '@/components/SendButton'
import { registerUserMutation } from '@/domains/auth/graphql'
import { getData } from '@/helpers/graphql'
import { mapActions } from 'vuex'

export default {
  name: 'Register',
  components: {
    LogoCard,
    FormCard,
    FormErrorMessage,
    SendButton
  },
  registerUserMutation,
  data: () => ({
    valid: false,
    input: {
      name: '',
      email: '',
      password: '',
      phone: ''
    },
    form: [
      {
        label: 'Nome',
        model: 'name',
        icon: 'fa-user',
        rules: 'required|alpha_spaces'
      },
      {
        label: 'Email',
        model: 'email',
        icon: 'fa-envelope',
        rules: 'required|email'
      },
      {
        label: 'Telefone',
        model: 'phone',
        icon: 'fa-phone',
        rules: 'required',
        mask: '(##) ##### - ####',
        type: 'tel'
      },
      {
        label: 'Senha',
        model: 'password',
        icon: 'fa-lock',
        type: 'password',
        rules: 'required|min:6'
      }
    ]
  }),
  methods: {
    ...mapActions('auth', ['setTokens']),
    changeRoute (name) {
      this.$router.push({ name })
    },
    submitSuccess (result) {
      return Promise
        .resolve(result)
        .then(getData('registerUser'))
        .then(this.setTokens)
        .then(() => this.changeRoute('home.bars'))
    },
    handleError (error) {
      this.$refs.formErrorMessage.handleError(error)
    }
  }
}
</script>
