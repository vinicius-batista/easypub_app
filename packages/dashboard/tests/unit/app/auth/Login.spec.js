import { fireEvent, cleanup } from '@testing-library/vue'
import Login from '@/app/auth/views/Login'
import Register from '@/app/auth/views/Register'
import App from '@/App'
import renderWithPlugins from '../../helpers/renderWithPlugins'

const routes = [
  { path: '/auth/register', name: 'auth.register', component: Register },
  { path: '/auth/login', name: 'auth.login', component: Login },
]

describe('Login.vue', () => {
  afterEach(cleanup)

  test('provide an incorret email and small password should show error messages', async () => {
    const { getByTestId, getByText, html } = renderWithPlugins(Login)

    const emailInput = getByTestId('email')
    await fireEvent.update(emailInput, 'jose.com')

    const passwordInput = getByTestId('password')
    await fireEvent.update(passwordInput, '12345')

    getByText('O campo Email deve ser um email válido')
    getByText('O campo Senha deve conter pelo menos 6 caracteres')

    expect(html).toMatchSnapshot()
  })

  test('provide a corret email and password should not show error messages', async () => {
    const { getByTestId, getByText, html } = renderWithPlugins(Login)

    const emailInput = getByTestId('email')
    await fireEvent.update(emailInput, 'jose@jose.com')

    const passwordInput = getByTestId('password')
    await fireEvent.update(passwordInput, '123456')

    expect(() =>
      getByText('O campo Email deve ser um email válido')
    ).toThrowError()
    expect(() =>
      getByText('O campo Senha deve conter pelo menos 6 caracteres')
    ).toThrowError()

    expect(html).toMatchSnapshot()
  })

  test('click action-text should change route to register', async () => {
    const { getByTestId, html } = renderWithPlugins(
      App,
      {
        routes,
      },
      (vue, store, router) => router.push('/auth/login')
    )

    const actionText = getByTestId('action-text')
    await fireEvent.click(actionText)
    getByTestId('register-page')

    expect(html).toMatchSnapshot()
  })
})
