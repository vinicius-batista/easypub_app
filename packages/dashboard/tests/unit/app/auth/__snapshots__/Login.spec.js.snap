// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.vue click action-text should change route to register 1`] = `[Function]`;

exports[`Login.vue provide a corret email and password should not show error messages 1`] = `[Function]`;

exports[`Login.vue provide an incorret email and small password should show error messages 1`] = `[Function]`;
